android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -705,42 +705,40 @@ constructor(
    * This function saves [QuestionnaireResponse] as draft if any of the [QuestionnaireResponse.item]
    * has an answer.
    */
-  fun saveDraftQuestionnaire(
+  suspend fun saveDraftQuestionnaire(
```

Comment:
this prevents a racing condition where the QR is saved first, but the process below is not done yet.